scripts/processPrivateApiData/grouping.ts

```diff
@@ -0,0 +1,76 @@
+import { PrivateApiUsageDto } from './dto';
+
+export type Group<Key, Value> = {
+  key: Key;
+  values: Value[];
+};
+
+export function grouped<Key, Value>(
+  values: Value[],
+  keyForValue: (value: Value) => Key,
+  areKeysEqual: (key1: Key, key2: Key) => boolean,
+) {
+  const result: Group<Key, Value>[] = [];
```

Comment:
could we use `Map<Key, Value[]>` here to simplify the types a bit?

Reply:
Your question made me wonder why I had even created `Group` in the first place instead of just using `Map`! But I think it's because I wanted to use custom logic for determining whether two keys were equal, and I thought that using `Map` might make things confusing. Honestly, given that this is only a script, I don't think it's worth spending much time worrying about.